config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java

```diff
@@ -3271,6 +3264,167 @@ public Boolean completeMd5() {
+                                                final String appName, final String ids) {
+        String tenantTmp = StringUtils.isBlank(tenant) ? StringUtils.EMPTY : tenant;
+        String sql = "select data_id,group_id,tenant_id,app_name,content,type from config_info";
+        StringBuilder where = new StringBuilder(" where ");
+        List<String> paramList = new ArrayList<>();
+        if(StringUtils.isNotBlank(ids)){
+            where.append(" id in (").append(ids).append(") ");
+        } else {
+            where.append(" tenant_id=? ");
+            paramList.add(tenantTmp);
+            if (StringUtils.isNotBlank(group)) {
+                where.append(" and group_id=? ");
+                paramList.add(group);
+            }
+            if (StringUtils.isNotBlank(appName)) {
+                where.append(" and app_name=? ");
+                paramList.add(appName);
+            }
+        }
+        try {
+            return this.jt.query(sql + where, paramList.toArray(), CONFIG_INFO_ROW_MAPPER);
+        } catch (CannotGetJdbcConnectionException e) {
+            fatalLog.error("[db-error] " + e.toString(), e);
+            throw e;
+        }
+    }
+
+    /**
+     * batch operation,insert or update
+     * the format of the returned:
+     * succCount: number of successful imports
+     * skipCount: number of import skips (only with skip for the same configs)
+     * failData: import failed data (only with abort for the same configs)
+     * skipData: data skipped at import  (only with skip for the same configs)
+     */
+    public Map<String, Object> batchInsertOrUpdate(List<ConfigInfo> configInfoList, String srcUser, String srcIp,
+                                                   Map<String, Object> configAdvanceInfo, Timestamp time, boolean notify, SameConfigPolicy policy) throws NacosException {
+        int succCount = 0;
+        int skipCount = 0;
+        List<Map<String, String>> failData = null;
+        List<Map<String, String>> skipData = null;
+
+        for (int i = 0; i < configInfoList.size(); i++) {
+            ConfigInfo configInfo = configInfoList.get(i);
+            try {
+                ParamUtils.checkParam(configInfo.getDataId(), configInfo.getGroup(), "datumId", configInfo.getContent());
+            } catch (NacosException e) {
+                defaultLog.error("data verification failed", e);
+                throw e;
+            }
+            ConfigInfo configInfo2Save = new ConfigInfo(configInfo.getDataId(), configInfo.getGroup(),
+                configInfo.getTenant(), configInfo.getAppName(), configInfo.getContent());
+
+            // simple judgment of file type based on suffix
+            String type = null;
+            if (configInfo.getDataId().contains(".")) {
+                String extName = configInfo.getDataId().substring(configInfo.getDataId().lastIndexOf(".") + 1).toLowerCase();
+                switch (extName) {
```

Comment:
There are two issues here：
1. What happens when the type is empty？
2. Plz use enum instead of switch
```suggestion
              // switch (extName) {
               // ....
               // }
                Optional<FileTypeEnum> fileTypeEnum = Enums.getIfPresent(FileTypeEnum.class, extName);
               public enum FileTypeEnum {
    YAML, JSON, PROPERTIES;
}

```